evalml/automl/utils.py

```diff
@@ -53,15 +53,17 @@ def make_data_splitter(X, y, problem_type, problem_configuration=None, n_splits=
     """
     problem_type = handle_problem_types(problem_type)
     data_splitter = None
+    if X.shape[0] > _LARGE_DATA_ROW_THRESHOLD:
+        return TrainingValidationSplit(test_size=_LARGE_DATA_PERCENT_VALIDATION, shuffle=True)
+
     if problem_type == ProblemTypes.REGRESSION:
-        data_splitter = KFold(n_splits=n_splits, random_state=random_state, shuffle=shuffle)
+        return KFold(n_splits=n_splits, random_state=random_state, shuffle=shuffle)
     elif problem_type in [ProblemTypes.BINARY, ProblemTypes.MULTICLASS]:
-        data_splitter = StratifiedKFold(n_splits=n_splits, random_state=random_state, shuffle=shuffle)
+        return StratifiedKFold(n_splits=n_splits, random_state=random_state, shuffle=shuffle)
     elif is_time_series(problem_type):
         if not problem_configuration:
             raise ValueError("problem_configuration is required for time series problem types")
-        data_splitter = TimeSeriesSplit(n_splits=n_splits, gap=problem_configuration.get('gap'),
-                                        max_delay=problem_configuration.get('max_delay'))
-    if X.shape[0] > _LARGE_DATA_ROW_THRESHOLD:
-        data_splitter = TrainingValidationSplit(test_size=_LARGE_DATA_PERCENT_VALIDATION, shuffle=True)
+        return TimeSeriesSplit(n_splits=n_splits, gap=problem_configuration.get('gap'),
+                               max_delay=problem_configuration.get('max_delay'))
+
     return data_splitter
```

Comment:
if we're not using `data_splitter`, can't we remove line 55 and just return `None` here?